design/ceph-config-updates.md

```diff
@@ -0,0 +1,258 @@
+
+Test (especially integration tests) may need to specify `osd pool default size = 1` and
+`osd pool default min size = 1` to support running clusters with only one osd. Test environments
+would have a means of doing this fairly easily using the config override capability. These values
+should not be set to these low values for production clusters, as they may allow admins to create
+their own pools which are not fault tolerant accidentally.
+
+There is an option to set these values automatically for clusters which run with only one osd or to
+set this value for clusters with a number of osds less than the default programmatically within
+Rook's operator; however, this adds an additional amount of code flow complexity which is
+unnecessary except in the integration test environments or in minimal demo environments. A middle
+ground proposed herein is to add `osd pool default {,min} size = 1` overrides to the example cluster
+CRD so that users "just trying out Rook" still get today's easy experience but where they can be
+easily removed for production clusters that should not run with potentially dangerous settings.
+
+### Changes to Ceph mons
+
+The current method of starting mons where `mon-a` has an initial member of `a`, `mon-b` initial
+members `a b`, `mon-c` initial members `a b c`, etc. Has worked so far but could result in a race
+condition. Mon cluster stability is important to Ceph, and it is critical for this PR that the mons'
+centrally-stored config is stable, so we here note that this behavior should be fixed such that the
+mon initial members are known before the first mon is bootstrapped to consider this proposal's work
+completed. Practically speaking, this will merely require the mon services to be started and have
+IP addresses before the mons are bootstrapped.
+
+Additionally, generating the monmap during mon daemon initialization is unnecessary if `--mon-host`
+is set for the `ceph-mon --mkfs` command.
+
+Creating `/var/lib/ceph/mon-<ID>/data/kv_backend` is no longer necessary in Luminous and
+can be removed.
+
+
+## Planning changes
+
+This proposal herein makes the suggestion that the changes be done with a new PR for each daemon
+starting with the mons, as the mons are most affected. After the mons are done, the remaining 4
+daemons can be done in parallel.
+
+Once all 5 daemons are complete, there will likely be a need to refactor the codebase to remove any
+vestigial remnants of the old config design which have been left. It will also be a good time to
+look for any additional opportunities to reduce code duplication by teasing repeated patterns out
+into shared modules.
+
+Another option is to modify all 5 daemons such that support is focused on Luminous, and the final
+clean-up stage could be a good time to introduce support for Mimic and its new centralized mon KV
+all at once.
+
+### Supporting Luminous
+
+Luminous does not have the mon's centralized kv store for Ceph configs, so any config set in the mon
```

Comment:
Is it reasonable to support both luminous and the new changes at the same time? It seems doable, but will complicate the config code to support both. Perhaps in 1.0 we could start with the minimal config in ceph.conf and use cmd line params as much as possible? Then in 1.1 complete the transition if that's about the right timeline to deprecate support for luminous?